wom/src/main/scala/wdl4s/wdl/Declaration.scala

```diff
@@ -92,10 +98,43 @@ object Declaration {
       ast
     )
   }
+
+  def buildWomNode(decl: Declaration): ErrorOr[DeclarationNode] = {
+
+    val inputName = decl.unqualifiedName
+
+    def declarationAsExpressionNode(wdlExpression: WdlExpression) = {
+      val womExpression = WdlWomExpression(wdlExpression, None)
+      for {
+        uninstantiatedExpression <- WdlWomExpression.graphNodeInputExpression(inputName, womExpression, decl)
+        expressionNode <- ExpressionNode.linkWithInputs(inputName, womExpression, uninstantiatedExpression.inputMapping)
+      } yield Left(expressionNode)
+    }
+
+    decl match {
+      case Declaration(opt: WdlOptionalType, _, None, _, _) => Valid(Right(OptionalGraphInputNode(inputName, opt)))
+      case Declaration(_, _, None, _, _) => Valid(Right(RequiredGraphInputNode(inputName, decl.wdlType)))
+      case Declaration(_, _, Some(expr), _, _) if expr.variableReferences.isEmpty => Valid(Right(OptionalGraphInputNodeWithDefault(inputName, decl.wdlType, WdlWomExpression(expr, None))))
+      case Declaration(_, _, Some(expr), _, _) => declarationAsExpressionNode(expr)
+    }
+  }
+
+  implicit class EnhancedDeclarationNode(val d: DeclarationNode) extends AnyVal {
+    def toGraphNode: GraphNode = d match {
+      case Left(expressionNode) => expressionNode
+      case Right(graphInputNode) => graphInputNode
```

Comment:
My only bone of contention with `fold` is for `Option`s, and even then I like the idea on paper due to type safety